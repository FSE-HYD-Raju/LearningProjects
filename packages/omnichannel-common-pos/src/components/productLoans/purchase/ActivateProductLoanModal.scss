.ActivateProductLoanModal {
	&-container {
		display: flex;
		flex-direction: column;
		margin-top: 0.625rem;
	}
	&-info-title-section {
		position: relative;
		margin: 0 1.875rem;
		i {
			position: absolute;
			top: 0.35rem;
			left: -1.75rem;
			font-size: 1rem;
			color: $primary;
		}
	}
	&-messages {
		width: 100%;
		display: flex;
		flex-wrap: wrap;
		&.ActivateProductLoanModal-confirmation-request {
			margin-top: 1.2rem;
		}
		span {
			width: 100%;
		}
	}
	&-loan-details-section {
		border-bottom: 0.0625rem solid $weasel-border-color-solid;
		padding: 0.8rem 1.875rem 1.5rem;
	}
	&-line {
		display: flex;
		align-items: center;
		width: 18.75rem;
		justify-content: space-between;
	}
	&-label {
		color: $gray-600;
		width: 10.25rem;
		padding: 0.25rem 0;
	}
	&-data {
		font-weight: $font-weight-bold;
	}
	&-confirmation-section {
		margin: 1.5rem 1.875rem 1rem;
		max-width: 27.5rem;
		input {
			margin-right: 0.5rem;
			font-size: 3rem;
		}
		label {
			margin-bottom: 0;
		}
	}
	&-bold {
		font-weight: $font-weight-bold;
	}
}

.ActivateProductLoanModal.OcModal {
	.modal-title {
		font-weight: $font-weight-bold !important;
	}
}

