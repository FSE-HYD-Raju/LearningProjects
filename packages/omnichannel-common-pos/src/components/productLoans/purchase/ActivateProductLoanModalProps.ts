import { InitializeAddonConfig } from "../../../redux/services/AddonService";
import { SimplePrice } from "../../../redux/types";

interface ActivateProductLoanModalOwnProps {
	msisdn: string;
}
interface ActivateProductLoanModalStateProps {
	loanedBalance: SimplePrice;
	fee: SimplePrice;
	totalCredit: SimplePrice;
	initializeAddonConfig: InitializeAddonConfig | undefined;
}
interface ActivateProductLoanModalActionProps {
	actions: {
		closeModal: () => void;
		confirm: (initializeAddonConfig: InitializeAddonConfig) => void;
	};
}
interface ActivateProductLoanModalProps
	extends ActivateProductLoanModalOwnProps,
		ActivateProductLoanModalStateProps,
		ActivateProductLoanModalActionProps {}
interface ActivateProductLoanModalState {
	agreedToTerms: boolean;
}
export {
	ActivateProductLoanModalOwnProps,
	ActivateProductLoanModalStateProps,
	ActivateProductLoanModalActionProps,
	ActivateProductLoanModalProps,
	ActivateProductLoanModalState
};
