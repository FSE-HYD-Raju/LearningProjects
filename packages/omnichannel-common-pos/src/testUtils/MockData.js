import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
