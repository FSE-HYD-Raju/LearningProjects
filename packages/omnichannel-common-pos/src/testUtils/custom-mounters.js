import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
