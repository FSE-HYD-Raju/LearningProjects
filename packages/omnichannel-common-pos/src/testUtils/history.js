import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
