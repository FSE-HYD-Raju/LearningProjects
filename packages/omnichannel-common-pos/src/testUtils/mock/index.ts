import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
