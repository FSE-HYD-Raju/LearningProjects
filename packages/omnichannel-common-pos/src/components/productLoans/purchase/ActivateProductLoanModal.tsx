import messages from "./activateProductLoan.messages";
import cssns from "../../../utils/cssnsConfig";
import FormattedMessage from "../../../channelUtils/FormattedMessage";
import OcCurrency from "../../ocComponents/OcCurrency";
import { ActivateProductLoanModalProps, ActivateProductLoanModalState } from "./ActivateProductLoanModalProps";
import OcModal from "../../ocComponents/OcModal";
import { PureComponent } from "react";

const { React } = cssns("ActivateProductLoanModal");

class ActivateProductLoanModal extends PureComponent<ActivateProductLoanModalProps, ActivateProductLoanModalState> {

	constructor(props: ActivateProductLoanModalProps) {
		super(props);
		this.state = {
			agreedToTerms: false
		};
	}

	getContent() {
		const { msisdn, fee, loanedBalance, totalCredit } = this.props;
		return (
			<div className="container">
				<div className="info-title-section">
					<i className="fas fa-info-circle" />
					<div className="messages">
						<FormattedMessage
							{...messages.aboutToTakeLoan}
							values={{
								msisdn: <span className="bold">{msisdn}</span>
							}}
						/>
						<FormattedMessage {...messages.loanDetailsHeader} />
					</div>
				</div>
				<div className="loan-details-section">
					<div className="line">
						<div className="label">
							<FormattedMessage {...messages.loanedBalance} />
						</div>
						<div className="data">
							<OcCurrency cost={loanedBalance.taxFreeAmount} currency={loanedBalance.currency} />
						</div>
					</div>
					<div className="line">
						<div className="label">
							<FormattedMessage {...messages.loanFee} />
						</div>
						<div className="data">
							<OcCurrency cost={fee.taxFreeAmount} currency={fee.currency} />
						</div>
					</div>
					<div className="line">
						<div className="label">
							<FormattedMessage {...messages.productLoanTotalCredit} />
						</div>
						<div className="data">
							<OcCurrency cost={totalCredit.taxFreeAmount} currency={totalCredit.currency} />
						</div>
					</div>
				</div>
				<div className="confirmation-section">
					<div className="custom-checkbox custom-control">
						<input
							className="custom-control-input"
							type="checkbox"
							id="loan-terms-and-conditions"
							name="loan-terms-and-conditions"
							value="loan-terms-and-conditions"
							checked={this.state.agreedToTerms}
							onChange={this.onInputChange}
						/>
						<label className="custom-control-label" htmlFor="loan-terms-and-conditions">
							<FormattedMessage {...messages.termsAgreementConfirmation} />
						</label>
					</div>
					<div className="messages confirmation-request">
						<FormattedMessage {...messages.addingLoanConfirmationRequest} />
					</div>
				</div>
			</div>
		);
	}

	onInputChange = () => {
		this.setState({
			agreedToTerms: !this.state.agreedToTerms
		});
	};
	onConfirm = () => {
		const { actions, initializeAddonConfig } = this.props;
		if (!initializeAddonConfig) {
			return;
		}
		actions.confirm(initializeAddonConfig);
	};
	render() {
		const { actions, initializeAddonConfig } = this.props;
		return (
			<OcModal
				className="ActivateProductLoanModal"
				showModal={true}
				largeModal={true}
				title={<FormattedMessage {...messages.activateProductLoan} />}
				onClose={actions.closeModal}
				okDisabled={!this.state.agreedToTerms}
				onOk={this.onConfirm}
				okButtonLabel={<FormattedMessage {...messages.productLoanModalConfirm} />}
			>
				{this.getContent()}
			</OcModal>
		);
	}
}

export default ActivateProductLoanModal;
