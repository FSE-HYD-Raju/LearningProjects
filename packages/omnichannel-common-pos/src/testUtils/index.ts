import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
