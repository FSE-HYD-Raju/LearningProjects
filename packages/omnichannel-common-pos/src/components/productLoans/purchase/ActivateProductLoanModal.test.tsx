import * as React from "react";
import { mountWithContext, shallowWithContext, TestUtils } from "../../../testUtils";
import ActivateProductLoanModal from "./ActivateProductLoanModal";
import { ActivateProductLoanModalProps } from "./ActivateProductLoanModalProps";

describe("ActivateProductLoanModal", () => {
	let props: ActivateProductLoanModalProps;
	beforeEach(() => {
		props = {
			actions: { closeModal: () => {}, confirm: () => {} },
			msisdn: "+38 099 1234 567",
			initializeAddonConfig: undefined,
			loanedBalance: {
				taxFreeAmount: 25,
				currency: "EUR"
			},
			fee: {
				taxFreeAmount: 5,
				currency: "EUR"
			},
			totalCredit: {
				taxFreeAmount: -30,
				currency: "EUR"
			}
		};
	});

	it("succeeds at shallow mount with minimum props", () => {
		const wrapper = shallowWithContext(<ActivateProductLoanModal {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimum props", () => {
		mountWithContext(<ActivateProductLoanModal {...props} />);
	});

	it("enables Confirm button when input is checked", () => {
		const { getModalContents } = TestUtils;
		const wrapper = mountWithContext(<ActivateProductLoanModal {...props} />);
		let modalContents = getModalContents(wrapper);
		modalContents.find("input").simulate("change", { target: { value: true } });
		modalContents = getModalContents(wrapper);
		const btn = modalContents.find("#oc-modal-action-button").hostNodes();
		expect(btn.prop("disabled")).toEqual(false);
	});
});
