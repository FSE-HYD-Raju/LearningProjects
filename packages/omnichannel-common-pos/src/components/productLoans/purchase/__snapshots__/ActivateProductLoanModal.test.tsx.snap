// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivateProductLoanModal succeeds at shallow mount with minimum props 1`] = `
<OcModal
  className="ActivateProductLoanModal"
  hideCloseButton={false}
  largeModal={true}
  okButtonLabel={
    <FormattedMessageWrapper
      defaultMessage="Confirm"
      description="Activate product loan modal, confirm"
      id="activate-product-loan-modal-confirm"
      wrapperType="wrapper"
    />
  }
  okDisabled={true}
  onClose={[Function]}
  onOk={[Function]}
  showModal={true}
  title={
    <FormattedMessageWrapper
      defaultMessage="Activate Product Loan"
      description="Activate product loan modal header"
      id="activate-product-loan-header"
      wrapperType="wrapper"
    />
  }
>
  <div
    className="ActivateProductLoanModal-container"
  >
    <div
      className="ActivateProductLoanModal-info-title-section"
    >
      <i
        className="fas fa-info-circle"
      />
      <div
        className="ActivateProductLoanModal-messages"
      >
        <FormattedMessageWrapper
          defaultMessage="You are about to take loan for {msisdn}"
          description="Activate product loan modal, intention reminder"
          id="about-to-take-loan-reminder"
          values={
            Object {
              "msisdn": <span
                className="ActivateProductLoanModal-bold"
              >
                +38 099 1234 567
              </span>,
            }
          }
          wrapperType="wrapper"
        />
        <FormattedMessageWrapper
          defaultMessage="These are the details of the loan: "
          description="Activate product loan modal, loan details header text"
          id="loan-details-header-text"
          wrapperType="wrapper"
        />
      </div>
    </div>
    <div
      className="ActivateProductLoanModal-loan-details-section"
    >
      <div
        className="ActivateProductLoanModal-line"
      >
        <div
          className="ActivateProductLoanModal-label"
        >
          <FormattedMessageWrapper
            defaultMessage="Loaned balance"
            description="Activate product loan modal, loaned balance"
            id="activate-product-loan-balance"
            wrapperType="wrapper"
          />
        </div>
        <div
          className="ActivateProductLoanModal-data"
        >
          <OcCurrency
            cost={25}
            currency="EUR"
          />
        </div>
      </div>
      <div
        className="ActivateProductLoanModal-line"
      >
        <div
          className="ActivateProductLoanModal-label"
        >
          <FormattedMessageWrapper
            defaultMessage="Loan fee"
            description="Activate product loan modal, loan fee"
            id="activate-product-loan-fee"
            wrapperType="wrapper"
          />
        </div>
        <div
          className="ActivateProductLoanModal-data"
        >
          <OcCurrency
            cost={5}
            currency="EUR"
          />
        </div>
      </div>
      <div
        className="ActivateProductLoanModal-line"
      >
        <div
          className="ActivateProductLoanModal-label"
        >
          <FormattedMessageWrapper
            defaultMessage="Total credit"
            description="Activate product loan modal, total credit"
            id="activate-product-loan-total-credit"
            wrapperType="wrapper"
          />
        </div>
        <div
          className="ActivateProductLoanModal-data"
        >
          <OcCurrency
            cost={-30}
            currency="EUR"
          />
        </div>
      </div>
    </div>
    <div
      className="ActivateProductLoanModal-confirmation-section"
    >
      <div
        className="custom-checkbox custom-control"
      >
        <input
          checked={false}
          className="custom-control-input"
          id="loan-terms-and-conditions"
          name="loan-terms-and-conditions"
          onChange={[Function]}
          type="checkbox"
          value="loan-terms-and-conditions"
        />
        <label
          className="custom-control-label"
          htmlFor="loan-terms-and-conditions"
        >
          <FormattedMessageWrapper
            defaultMessage="I have read terms and conditions and agree to them"
            description="Activate product loan modal, terms and conditions agreement confirmation"
            id="terms-agreement-confirmation"
            wrapperType="wrapper"
          />
        </label>
      </div>
      <div
        className="ActivateProductLoanModal-messages ActivateProductLoanModal-confirmation-request"
      >
        <FormattedMessageWrapper
          defaultMessage="Please confirm that you want to add the loan to your subscription by clicking \\"Confirm\\"."
          description="Adding loan confirmation request message"
          id="adding-loan-confirmation-request-message"
          wrapperType="wrapper"
        />
      </div>
    </div>
  </div>
</OcModal>
`;
